packages/@lwc/integration-karma/test/light-dom/lifecycle/index.spec.js

```diff
@@ -34,149 +65,192 @@ describe('standard slotting', () => {
+
+                // Initial connection
+                expect(window.timingBuffer).toEqual([
+                    '0:connectedCallback',
+                    '1:connectedCallback',
+                    '2:connectedCallback',
+                ]);
+
+                resetTimingBuffer();
+
+                // Trigger vdom diffing
+                elm.showTop = true;
+                await Promise.resolve();
+
+                expect(window.timingBuffer).toEqual([
+                    '3:connectedCallback',
+                    '4:connectedCallback',
+                    '5:connectedCallback',
+                    '2:disconnectedCallback',
+                    '0:disconnectedCallback',
+                    '1:disconnectedCallback',
+                ]);
+
+                resetTimingBuffer();
+
+                // Trigger vdom diffing from top level
+                elm.topTop = 'bottom';
+                elm.topBottom = 'top';
+                await Promise.resolve();
+
+                expect(window.timingBuffer).toEqual([
+                    '6:connectedCallback',
+                    '5:disconnectedCallback',
+                    '7:connectedCallback',
+                    '3:disconnectedCallback',
+                ]);
+
+                resetTimingBuffer();
+
+                const { topSlot } = extractDataIds(elm);
+                // Trigger vdom diffing from forwarded slot level
+                topSlot.top = 'top';
+                topSlot.bottom = 'bottom';
+                await Promise.resolve();
+
+                expect(window.timingBuffer).toEqual([
+                    '8:connectedCallback',
+                    '6:disconnectedCallback',
+                    '9:connectedCallback',
+                    '7:disconnectedCallback',
+                ]);
+
+                resetTimingBuffer();
+            };
+
+            it('invokes lifecycle methods in correct order', async () => {
+                await setup();
+
+                // vdom diffing after changing vnodes
+                elm.showTop = false;
+                await Promise.resolve();
+
+                expect(window.timingBuffer).toEqual([
+                    '10:connectedCallback',
+                    '11:connectedCallback',
+                    '12:connectedCallback',
+                    '8:disconnectedCallback',
+                    '9:disconnectedCallback',
+                    '4:disconnectedCallback',
+                ]);
+            });
+
+            it('invokes disconnectedCallback after removing entire element', async () => {
+                await setup();
+
+                // remove entire element
+                document.body.removeChild(elm);
+                await Promise.resolve();
+
+                expect(window.timingBuffer).toEqual([
+                    '8:disconnectedCallback',
+                    '9:disconnectedCallback',
+                    '4:disconnectedCallback',
+                ]);
+            });
```

Comment:
I didn't strictly need to add any `remove entire element` tests here, but I wanted to cover all my bases.